cmd/instance_delete.go

```diff
@@ -36,33 +35,64 @@ func (c *instanceDeleteCmd) cmdPreRun(cmd *cobra.Command, args []string) error {
 }
 
 func (c *instanceDeleteCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, c.Zone))
+	ctx := gContext
+	client, err := switchClientZoneV3(
+		ctx,
+		globalstate.EgoscaleV3Client,
+		v3.ZoneName(c.Zone),
+	)
+	if err != nil {
+		return err
+	}
 
-	instance, err := globalstate.EgoscaleClient.FindInstance(ctx, c.Zone, c.Instance)
+	instances, err := client.ListInstances(ctx)
 	if err != nil {
-		if errors.Is(err, exoapi.ErrNotFound) {
-			return fmt.Errorf("resource not found in zone %q", c.Zone)
-		}
 		return err
 	}
 
-	if !c.Force {
-		if !askQuestion(fmt.Sprintf("Are you sure you want to delete instance %q?", c.Instance)) {
-			return nil
+	instanceToDelete := []v3.UUID{}
+	for _, i := range c.Instances {
+		instance, err := instances.FindListInstancesResponseInstances(i)
+		if err != nil {
+			if !c.Force {
+				return err
+			}
+
```

Comment:
I think it would be useful to print error (to `stderr`) before `continue`. Flag `--force` is to not show confirmation dialog, but it doesn't have to be silent command and we can still print error.